report-ng/app/src/components/timings/test-timings/test-timings.ts

```diff
@@ -0,0 +1,373 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {autoinject, bindable, observable} from 'aurelia-framework';
+import {NavigationInstruction, RouteConfig} from "aurelia-router";
+import {AbstractViewModel} from "../../abstract-view-model";
+import * as echarts from 'echarts';
+import {EChartsOption} from 'echarts';
+import "./test-timings.scss";
+import {ExecutionStatistics} from "services/statistic-models";
+import {MethodDetails, StatisticsGenerator} from "services/statistics-generator";
+import moment from "moment";
+import {data} from "../../../services/report-model";
+import {StatusConverter} from "../../../services/status-converter";
+import MethodType = data.MethodType;
+import IMethodContext = data.MethodContext;
+
+@autoinject()
+export class TestTimings extends AbstractViewModel {
+    private _chart: echarts.ECharts;
+    private _executionStatistics: ExecutionStatistics
+    private _option: EChartsOption;
+    private _hasEnded = false;
+    private _methodDetails: MethodDetails[];
+    private _labels: string[];
+    private _sectionValues: number[];
+    private _data: number[];
+    private _bars: IDurationBar[];
+    private _loading = false;
+    private _searchRegexp: RegExp;
+    private _inputValue = '';
+    private _methodId: string;
+    @bindable private _rangeNum;
+    private _rangeOptions = ['5', '10', '15', '20'];
+    private _showConfigurationMethods: boolean = null;
+    private _lookUpOptions;
+    @observable private selectedOptionId;
+    private _methodNameInput;
+    private static readonly TEST_NUMBER_LIMIT = 20;
+
+    constructor(
+        private _statusConverter: StatusConverter,
+        private _statisticsGenerator: StatisticsGenerator,
+    ) {
+        super();
+        this._bars = [];
+        this._lookUpOptions = [];
+        this._option = {};
+    }
+
+    activate(params: any, routeConfig: RouteConfig, navInstruction: NavigationInstruction) {
+        super.activate(params, routeConfig, navInstruction);
+        if (!this.queryParams.rangeNum) {
+            this._rangeNum = '10';
+            this.queryParams.rangeNum = parseInt(this._rangeNum);
+        }
+        if (params.config) {
+            this._showConfigurationMethods = !!params.config.toLowerCase();
+        }
+        this._getLookUpOptions()
+        this._filter();
+    }
+
+    selectedOptionIdChanged(n: string) {
+        if (n) {
+            this._methodId = n;
+            this.queryParams.methodId = n
+            this.updateUrl(this.queryParams);
+            this._highlightData()
+        }
+    }
+
+    selectionChanged() { // called when input from lookup text field is deleted manually
+        this._setChartOption(); // overwrites color highlighting (reset)
+        if (this._inputValue.length == 0) {
+            this._methodId = undefined;
+            this.queryParams.methodId = undefined;
```

Comment:
Consider deleting it, because it could result in an empty query parameter like `http://xxx/view?methodId=`
```typescript
delete (this.queryParams.methodId)
```